services/database/user.ts

```diff
@@ -35,4 +37,145 @@ async function getUsers() {
     return users;
 }
 
-export { getUser, getUsers };
+/**
+ * Updates a user with the data corresponding to userInput
+ * @param userInput - data for the updated user
+ * @returns prisma User with updated information
+ */
+
+async function updateUser(userInput: UserInput): Promise<User> {
+    /*
+    Flow:
+    - get role and role data from UserInput
+    - create/update the role record
+    - update the user record
+    - return the user, including the role records
+    */
+    const roleData = userInput.roleData;
+    if (!roleData.id) {
+        roleData.id = userInput.id;
```

Comment:
Kinda confused, why does userInput and roleData both have id fields that mean the same thing? (i might be fried LOL)

Reply:
Good point, these were there because `roleData` previously needed the `id` but it doesn't anymore. I'll get rid of it